truffleruby/src/main/c/openssl/ossl_asn1.c

```diff
@@ -845,7 +845,7 @@ int_ossl_asn1_decode0_prim(unsigned char **pp, long length, long hlen, int tag,
     *pp += hlen + length;
     *num_read = hlen + length;
 
-    if (tc == sUNIVERSAL && tag < ossl_asn1_info_size && ossl_asn1_info_klass[tag]) { // ossl_asn1_info[tag].klass) {
+    if (tc == sUNIVERSAL && tag < ossl_asn1_info_size && truffle_is_truffle_object(ossl_asn1_info_klass[tag])) { // ossl_asn1_info[tag].klass) {
```

Comment:
@chrisseaton I think it's trying to cast the `OpenSSL::ASN1::Boolean` in the error to an `int` to do an int comparison.  I think `NULL` should work fine for truthiness. Alternatively, `NIL_P` might work for this (since `Qnil` should `= NULL` should `= 0` but I'm not sure yet..

@eregon I tried ` != NULL` but it resulted in the same error. I've added a comment.